packages/rum/src/getActionNameFromElement.ts

```diff
@@ -1,16 +1,48 @@
 export function getActionNameFromElement(element: Element): string {
   // Proceed to get the action name in two steps:
-  // * first, use strategies that are known to return good results. Those strategies will be used on
+  // * first, get the name programmatically, explicitely defined by the user.
+  // * then, use strategies that are known to return good results. Those strategies will be used on
   //   the element and a few parents, but it's likely that they won't succeed at all.
   // * if no name is found this way, use strategies returning less accurate names as a fallback.
   //   Those are much likely to succeed.
   return (
+    getActionNameFromElementProgrammatically(element) ||
     getActionNameFromElementForGetters(element, priorityGetters) ||
     getActionNameFromElementForGetters(element, fallbackGetters) ||
     ''
   )
 }
 
+/**
+ * Get the action name from the attribute 'data-dd-action-name' on the element or any of its parent.
+ */
+const PROGRAMMATIC_ATTRIBUTE = 'data-dd-action-name'
+function getActionNameFromElementProgrammatically(targetElement: Element) {
+  let elementWithAttribute
+  // We don't use getActionNameFromElementForGetters here, because we want to consider all parents,
```

Comment:
What is the reasoning for considering all parents and not some parents? or only the target element?

Reply:
> What is the reasoning for considering all parents and not some parents?
Contrary to other strategies, where we try our best to be smart by using heuristics based on how interfaces are built in general, the programmatic one should be dumb, trivial and unsurprising. This allows users to experiment how they want, without limit.

> or only the target element?
I find it less convenient, as a button (or other control) is often composed of multiple elements, so every elements would need to have the attribute to consider the button as a whole.

Reply:
OK, we should probably also document this behavior in the API then.

Reply:
I already wrote
> on clickable elements (or any of their parents)
Do you think I should elaborate?

Reply:
I quickly checked and missed it 🤦 
A second example with a parent case could be clearer

Reply:
Ok I'll add it!